addons/stock/views/stock_scrap_views.xml

```diff
@@ -61,6 +61,7 @@
                                 <field name="origin"/>
                                 <field name="date_expected"/>
                                 <field name="picking_id" attrs="{'invisible': [('picking_id', '=', False)]}"/>
+                                <field name="company_id" groups="base.group_multi_company"/>
```

Comment:
As you add the company_id on the scrap. We should set a domain on location_id and scrap_location_id to restrict on the scrap's company. 

The default fonction in the model (_get_default_scrap_location_id and _get_default_location_id) should take location of the company as well (or location without company).